uaclient/actions.py

```diff
@@ -159,8 +159,29 @@ def attach_with_token(
         contract_token=token, attachment_dt=attached_at
     )
 
-    machine_token_file.write(new_machine_token)
+    current_series = system.get_release_info().series
+    only_series = (
+        machine_token_file.entitlements()
+        .get("support", {})
+        .get("entitlement", {})
+        .get("affordances", {})
+        .get("onlySeries", None)
+    )
+    if only_series:
+        allowed_release = system.get_distro_info(only_series)
+        if only_series != current_series:
+            machine_token_file.delete()
```

Comment:
Adding a new comment so it's more visible - I don't think we need this `delete` call anymore - right?

Reply:
We cannot call the detach api when we check for the onlySeries in the attach operation (due to lock issues), so I think we might want to keep this here as a way to detach?

Reply:
While checking the behave test failure, I realized that the test failed because we pushed this chunk of code before the machineTokenFile is written to. What this does is that the contract response (in the case of the test, the response_json) is not written to, so the `machine_token_file.entitlements()` function returns an empty dict.

Possible solution is to directly check the `new_machine_token` variable for the onlySeries or move the `machine_token_file.write()` operation to where it was earlier?

Reply:
@dheyay my idea when I first saw this was to get `onlySeries` from `new_machine_token`, yes. If you think it looks messy, maybe rename the variable to `contract_machine_token` or `received_machine_token` - although for me the current name explains it well and checking information there is completely fine